numba/npyufunc/parfor.py

```diff
@@ -358,12 +370,26 @@ def call_parallel_gufunc(lowerer, cres, gu_signature, outer_sig, expr_args, loop
     if config.DEBUG_ARRAY_OPT:
         print("parallel function = ", wrapper_name, cres)
 
-    if config.DEBUG_ARRAY_OPT:
-        cgutils.printf(builder, "loop_ranges = ")
-        for v in loop_ranges:
-            print("call_parallel_gufunc v = ", v, " ", lowerer.loadvar(v))
-            cgutils.printf(builder, "%d ", lowerer.loadvar(v))
-        cgutils.printf(builder, "\n")
+    # loadvars for loop_ranges
+    def load_range(v):
+        if isinstance(v, ir.Var):
+            return lowerer.loadvar(v.name)
+        else:
+            return context.get_constant(types.intp, v)
+
+    num_dim = len(loop_ranges)
+    for i in range(num_dim):
+        start, stop, step = loop_ranges[i]
+        start = load_range(start)
+        stop = load_range(stop)
+        assert(step == 1) # We do not support loop steps other than 1
```

Comment:
Thanks, I see. In which case should this perhaps raise an exception as an unsupported feature so it is clear what the issue is to the user? Also, a test to make sure this is caught would be good.

This also got me thinking more about slicing etc. This (slight mutation of one of the test cases) should raise as it's a violation of broadcast semantics (it does so without `parallel=True`):
```python
@njit(parallel=True)
def test_2d():
    X = np.ones((10,12))
    Y = np.zeros((10,12))
    return np.sum(X[::2] + Y) # step 2 creates invalid arrays for addition
```
instead it returns `nan`/junk.